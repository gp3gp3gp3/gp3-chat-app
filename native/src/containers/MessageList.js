import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import MessageList from '../components/MessageList'
import { fetchMessages } from '../actions'

export class MessageListContainer extends PureComponent {
  componentDidMount () {
    this.props.fetchMessages()
  }

  componentDidUpdate () {
    this.props.scrollToBottom()
  }

  render () {
    return (
      <MessageList
        {...this.props}
      />
    )
  }
}

const mapStateToProps = ({ messages, auth: { currentUser } }) => {
  return {
    messages,
    currentUser
  }
}

export default connect(mapStateToProps, { fetchMessages })(MessageListContainer)
