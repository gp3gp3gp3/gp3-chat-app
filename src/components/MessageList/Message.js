import React from 'react'
import './MessageList.css'

const Message = ({ message }) => (
  <span className='Message'>{message}</span>
)

export default Message
