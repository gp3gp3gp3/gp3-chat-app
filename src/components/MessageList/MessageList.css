.MessageList {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: 20px;
}

.current-user {
  align-self: flex-end;
}

.other-user {
  align-self: flex-start;
}

.green {
  background-color: greenyellow;
}

.blue {
  background-color: lightblue;
}

.Message {
  max-width: 50%;
  text-align: left;
  border-radius: 15px;
  padding: 15px;
  color: #222;
  margin: 10px 0;
  box-shadow: 0px 2px 2px rgba(0,0,0,0.25);
}

.Message-wrapper {
  margin-top: 30px;
  display: flex;
  flex-direction: column;
}

.Message-time {
  text-align: end;
  color: grey;
  font-size: 0.80em;
}

.message-enter {
  opacity: 0.01;
}

.message-enter.message-enter-active {
  opacity: 1;
  transition: opacity 500ms ease-in;
}

.message-leave {
  opacity: 1;
}

.message-leave.message-leave-active {
  opacity: 0.01;
  transition: opacity 300ms ease-in;
}
