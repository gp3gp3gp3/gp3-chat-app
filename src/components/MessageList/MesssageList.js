import React, { Component } from 'react'

class MessageList extends Component {
  render () {
    return (
      <div>
        MessageList Component
      </div>
    )
  }
}

export default MessageList
