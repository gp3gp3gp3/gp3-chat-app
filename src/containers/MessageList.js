import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import MessageList from '../components/MessageList/MessageList'
import { fetchMessages } from '../actions'

class MessageListContainer extends PureComponent {
  componentDidMount () {
    this.props.fetchMessages()
  }

  componentDidUpdate () {
    this.props.scrollToBottom()
  }

  render () {
    return (
      <MessageList
        {...this.props}
      />
    )
  }
}

const mapDispatchToProps = {
    fetchMessages
}

const mapStateToProps = ({ messages, auth: { currentUser } }) => {
  return {
    messages,
    currentUser
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(MessageListContainer)
