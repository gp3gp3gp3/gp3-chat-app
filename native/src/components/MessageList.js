import React from 'react'
import Message from './Message'

const MessageList = ({ messages, currentUser }) => (
  <div className='MessageList'>
    {/* <CSSTransitionGroup
      transitionName='message'
      transitionEnterTimeout={500}
      transitionLeaveTimeout={300}
    > */}
    {messages.map(message => <Message {...message} currentUser={currentUser} />)}
    {/* </CSSTransitionGroup> */}
  </div>
)

export default MessageList
