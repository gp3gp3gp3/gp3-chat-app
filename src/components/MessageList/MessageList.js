import React from 'react'
import Message from './Message'
import './MessageList.css'

const MessageList = ({ messages }) => (
  <div className='MessageList'>
    {messages.map(message => <Message {...message} />)}
  </div>
)

export default MessageList
