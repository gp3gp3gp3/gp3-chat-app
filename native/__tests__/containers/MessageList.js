/* eslint-env jest */
import 'react-native'
import { shallow } from 'enzyme'
import React from 'react'
import configureStore from 'redux-mock-store'
import MessageList from '../../src/containers/MessageList'

jest.mock('react-native-google-sign-in', () => {
  return {
    configure: jest.fn(),
    signInPromise: jest.fn()
  }
})

const middlewares = []
const mockStore = configureStore(middlewares)
const initialState = {
  auth: {
    authenticated: true,
    currentUser: {
      displayName: 'Test User',
      uid: '1234'
    }
  },
  messages: [
    {
      user: {
        displayName: 'Other user',
        uid: '12345'
      },
      message: 'Test Message',
      currentUser: {
        displayName: 'Test User',
        uid: '1234'
      },
      time: '12:30'
    },
    {
      user: {
        displayName: 'Other user',
        uid: '12345'
      },
      message: 'Test Message 2',
      currentUser: {
        displayName: 'Test User',
        uid: '1234'
      },
      time: '12:30'
    }
  ]
}

it('renders correctly with store', () => {
  const wrapper = shallow(
    <MessageList />,
    { context: { store: mockStore(initialState) } }
  )
  expect(wrapper).toMatchSnapshot()
})
