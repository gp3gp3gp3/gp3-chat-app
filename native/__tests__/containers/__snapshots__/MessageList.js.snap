// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with store 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(MessageListContainer) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "currentUser": Object {
        "displayName": "Test User",
        "uid": "1234",
      },
      "messages": Array [
        Object {
          "currentUser": Object {
            "displayName": "Test User",
            "uid": "1234",
          },
          "message": "Test Message",
          "time": "12:30",
          "user": Object {
            "displayName": "Other user",
            "uid": "12345",
          },
        },
        Object {
          "currentUser": Object {
            "displayName": "Test User",
            "uid": "1234",
          },
          "message": "Test Message 2",
          "time": "12:30",
          "user": Object {
            "displayName": "Other user",
            "uid": "12345",
          },
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "currentUser": Object {
          "displayName": "Test User",
          "uid": "1234",
        },
        "messages": Array [
          Object {
            "currentUser": Object {
              "displayName": "Test User",
              "uid": "1234",
            },
            "message": "Test Message",
            "time": "12:30",
            "user": Object {
              "displayName": "Other user",
              "uid": "12345",
            },
          },
          Object {
            "currentUser": Object {
              "displayName": "Test User",
              "uid": "1234",
            },
            "message": "Test Message 2",
            "time": "12:30",
            "user": Object {
              "displayName": "Other user",
              "uid": "12345",
            },
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
